File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        dict = {}
        for char in s:
            dict.update({char: (dict.setdefault(char, 0) + 1)})
        for char in t:
            count = dict.setdefault(char, 0)
            if count == 0: return False
            else: dict.update({char: count - 1})
        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        Map<Character, Integer> smap = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            smap.put(s.charAt(i), smap.getOrDefault(s.charAt(i), 0) + 1);
        }
        for (int i = 0; i< t.length(); i++) {
            int count = smap.getOrDefault(t.charAt(i), 0);
            if (count == 0) return false;
            else smap.put(t.charAt(i), count - 1);
        }
        return true;
    }
}

        